code/routers/country_page/country.js

```diff
@@ -99,28 +99,192 @@ router.get('/get-country-info', (req, res) => {
+            if(err) {
+              return res.status(countryCodes.insertOrganizations.organizationUpdateFail.status).send({
+                message: countryCodes.insertOrganizations.organizationUpdateFail.message
+              });
+            }else{
+              return res.status(countryCodes.insertOrganizations.success.status).send({
+                message: countryCodes.insertOrganizations.success.message
+              });
+            }
+          })
+        }
+      })
     }
   })
 })
 
 
+/*
+    Remove Organizations API
+    inputs:
+      [{
+      countryName: String,
+      orgName: String
+      }]
+
+    Expects all countryNames to be the same.
+*/
+router.post('/remove-organizations', (req, res) => {
+  
+  //Handling of multiple organizations, still expecting them to be from the same country
+  if(Array.isArray(req.body)){
+    var orgNames = [];
+    for (var i = 0; i < req.body.length; i++){
+      var curBody = req.body[i];
+      orgNames.push(curBody.orgName);
+    }
+    var orgQuery = {orgName: {$in: orgNames}};
+    var countryQuery = {name: req.body[0].countryName}; 
+  }else{
+    var orgQuery = {orgName: req.body.orgName}; 
+    var countryQuery = {name: req.body.countryName};  
+  }
+
+  Organization.find(orgQuery, function(err, organizations) {
+    if(err || organizations.length == 0){
+      return res.status(countryCodes.removeOrganizations.organizationsNotFound.status).send({
+        message: countryCodes.removeOrganizations.organizationsNotFound.message
+      });
+    }else{
+      var updatedOrgIDs = []
+      Country.find(countryQuery, function(err, country) {
+        if(err) {
+          return res.status(countryCodes.removeOrganizations.countryNotFound.status).send({
+            message: countryCodes.removeOrganizations.countryNotFound.message
+          });
+        }else{
+          var updatedOrgIDs = [];
+          var deleteOrgIDs = [];
+          for (var i = 0; i < organizations.length; i++) {
+            deleteOrgIDs.push(organizations[i]._id);
+          }
+
+          var prevOrgs = country[0].organizationIDs;
+          if(Array.isArray(prevOrgs)){
+            updatedOrgIDs = prevOrgs.filter(curOrg => {
+              var toKeep = true;
+              for (var i = 0; i < deleteOrgIDs.length; i++){
+                if(curOrg.equals(deleteOrgIDs[i])){
+                  toKeep = false;
+                  deleteOrgIDs.splice(i, 1); //remove organization since it's already been covered
+                  break;
+                }
+              }
+              return toKeep
+            })
+          }
+          // Normally doing this:
+
+          //country.organizationIDs = updatedOrgIDs;
+          //country.save();
+
+          // would be better to update country, but country is not a mongoose.Document
+          // this is likely due to how the data was imported on the mongoDB Atlas database
```

Comment:
1. Is this comment necessary?
2. Why is it not like this? If this is the optimal solution that we should try and make this work so that the code is simpler and thus less likely to cause bugs

Reply:
I see what you mean, it would be interesting to see this method work (and it is theoretically better, idk why, but mongoose documentation says so), but since this isn't a user-facing API, I think it isn't worth the effort to make it work. The problem with the `save()` function is that it exists for `mongoose.Documents`, but when doing a mongoose `Country.find(query, function(err, country){})`, country is not a document. I believe this has to do with how I imported the data to the database. Unfortunately I don't have the time to go back and try using mongoose (as opposed to directly calling `mongoimport`) to populate the database. 

I can delete the comment if you think I should, but I think it would be good to keep for the future team.

Reply:
The find method should return some json which I believe you can easily convert into a mongoose document doing something like this:

var organization = new Organization([JSON returned from find])

This assumes that it only returns one object, if it returns more you will have to loop over the list and convert each one.

Also, comments like this should be put in the wiki, it is bad practice to put long comments like this in the code. Generally the code should only have brief comments for things that aren't obvious. This should be put int the wiki and have a link to it in the code with a brief thing saying look at this wiki page for information. Before doing this try the thing above.